stake-pool/cli/src/main.rs

```diff
@@ -1464,18 +1464,15 @@ fn command_withdraw_stake(
             );
         }
 
-        // Use separate mutable variable because withdraw might create a new account
-        let stake_receiver = stake_receiver_param.unwrap_or_else(|| {
-            let stake_keypair = new_stake_account(
-                &config.fee_payer.pubkey(),
-                &mut instructions,
-                stake_account_rent_exemption,
-            );
-            let stake_pubkey = stake_keypair.pubkey();
-            total_rent_free_balances += stake_account_rent_exemption;
-            new_stake_keypairs.push(stake_keypair);
-            stake_pubkey
-        });
+        // Creating new account to split the stake into new account
+        let stake_keypair = new_stake_account(
+            &config.fee_payer.pubkey(),
+            &mut instructions,
+            stake_account_rent_exemption,
+        );
+        let stake_receiver = stake_keypair.pubkey();
+        total_rent_free_balances += stake_account_rent_exemption;
+        new_stake_keypairs.push(stake_keypair);
```

Comment:
You'll only want to do this if `stake_receiver_param.is_none()` or `stake_receiver_param.is_some()` and it's a stake account already -- you may as well add the checks done later into this.

Also, you'll need to move this up higher -- if someone is providing a delegated stake account, we should only do *one* withdrawal, from the corresponding stake account in the stake pool.  See where `withdraw_accounts` is being created higher up.  If it's a delegated stake account, we should use that vote account.  If a vote account is provided as well, but it doesn't match, then we should error.